app/controllers/tags_controller.rb

```diff
@@ -0,0 +1,8 @@
+class TagsController < ApplicationController
+  def republish
+    tag = Tag.find_by!(content_id: params[:tag_id])
+    authorise_user!("GDS Editor") if tag.is_a?(MainstreamBrowsePage)
```

Comment:
This could use a before_filter to be consistent with the other controllers. (it might make sense to move the `editing_lists_for_mainstream_browse_page?` method up to the application_controller as well to avoid duplication.)